src/component_selector.jl

```diff
@@ -0,0 +1,124 @@
+# Most of this is implemented by wrapping the IS versions, replacing
+# InfrastructureSystemsComponent with Component and SystemData with System
+
+"""
+    get_components(selector, sys; filterby = nothing)
+Get the components of the `System` that make up the `ComponentSelector`.
+ - `filterby`: optional filter function to apply after evaluating the `ComponentSelector`
+"""
+get_components(selector::ComponentSelector, sys::System; filterby = nothing) =
+    IS.get_components(selector, sys; filterby = filterby)
+
+"""
+    get_components(filterby, selector, sys)
+Get the components of the `System` that make up the `ComponentSelector`.
+ - `filterby`: optional filter function to apply after evaluating the `ComponentSelector`
+"""
+get_components(
+    filterby::Union{Nothing, Function},
+    selector::ComponentSelector,
+    sys::System,
+) =
+    get_components(selector, sys; filterby = filterby)
+
+# This would be cleaner if `IS.get_components === PSY.get_components` (see
+# https://github.com/NREL-Sienna/InfrastructureSystems.jl/issues/388)
+IS.get_components(selector::ComponentSelector, sys::System; filterby = nothing) =
+    IS.get_components(selector, sys.data; filterby = filterby)
+
+"""
+    get_component(selector, sys; filterby = nothing)
+Get the component of the `System` that makes up the `SingularComponentSelector`; `nothing`
+if there is none.
+ - `filterby`: optional filter function to apply after evaluating the `ComponentSelector`
+"""
+get_component(selector::SingularComponentSelector, sys::System; filterby = nothing) =
```

Comment:
Similar to above, these methods could be defined without kwargs or unions.

Reply:
I'm fine to add the method where the function is the first argument, but I'm hoping to keep the kwarg version as well — it makes more sense to me in the context where this would be used. Is that objectionable?

Reply:
I was outvoted; the `filterby` kwarg is no more (see the IS PR)